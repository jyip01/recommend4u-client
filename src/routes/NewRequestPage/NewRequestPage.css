.newRequestPage{
  display:flex;
  height:76vh;
}
  
.sideBar{
  width: 25%;
  height: 100%;
  padding-top:25px;
  background: url('../../images/recommend.jpg') no-repeat center right;
  padding-left:10px;
  padding-right:10px;
}
  
.backButton{
  width:100%;
  background-color:#384a62;
  color:white;
  display:flex;
  justify-content: center;
  align-items:center;
  padding:5px;
  text-decoration: none;
  font-weight: bold;
  border-radius: 2px;
  border:0px;
}

.backButton:hover{
  background-color: #8bccbc;
  border: 2px solid #384a62;
  cursor: pointer;
}
  
.mainBar{
  padding-top:25px;
  overflow:scroll;
  padding-left:20px;
  padding-right:20px;
  width:100%;
}

.mainBar h2{
  font-size:22px;
}

.inputBox {
  display:flex;
  flex-direction: column;
  padding-top:10px;
}

.inputBox input, .inputBox select, .inputBox textarea{
  border:1px solid black;
  border-radius: 6px;
  padding: 5px;
}

.inputBox label{
  padding-bottom: 5px;
}

.submitRequestButton{
  width:60%;
  margin-left: auto;
  margin-right: auto;
  background-color:#384a62;
  color:white;
  display:flex;
  justify-content: center;
  align-items:center;
  padding:5px;
  text-decoration: none;
  font-weight: bold;
  border-radius: 6px;
  margin-top:10px;

}

.submitRequestButton:hover{
  background-color: #8bccbc;
  border: 2px solid #384a62;
  cursor: pointer;
}

.newReqInput{
  width:100%;
  background-color:white;
}

@media only screen and (min-width: 600px) {
  .mainBar{
    padding-right:100px;
  }
}

@media only screen and (min-width: 800px) {
  .mainBar{
    padding-right:150px;
  }
}

@media only screen and (min-width: 1000px) {
  .mainBar{
    padding-right:400px;
  }
}
