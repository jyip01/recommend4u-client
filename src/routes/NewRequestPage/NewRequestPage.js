import React, { Component } from 'react'
import './NewRequestPage.css'
import RequestApiService from '../../services/request-api-services'
import UserService from '../../services/user-service'
import RequestListContext from '../../contexts/RequestListContext'
import Error from '../../components/Error/Error'

export default class NewRequestPage extends Component {
    static contextType = RequestListContext

    //used for redirecting to users requests page on form submit
    static defaultProps = {
        history: {
            push: () => {},
        }
    }
    
    state = {
        error: null
    }

    onSubmitSuccess = ()=>{
        //redirects to users requests page on successful form submission
        const { history } = this.props
        history.push('/requests/users')
    }

    //when form is submitted, posts new request and the GETs all requests and users request list
    handleSubmit = ev => {
        ev.preventDefault()
        const { product, category, moreInfo } = ev.target

        this.setState({ error: null })

        let userId = UserService.getUserToken()

        RequestApiService.postNewRequest(userId, product.value, category.value, moreInfo.value)
            .then(res=>{
                RequestApiService.getAllRequests()
                    .then(res=>{
                        this.context.setRequestList(res)
                        RequestApiService.getRequestsByUserId(userId)
                            .then(this.context.setUsersList)
                            .catch()
                    })
                    .catch()
                this.onSubmitSuccess()
            })
            .catch(this.context.setError)
    }

    componentDidMount =()=>{
        this.context.clearError()
    }
    
    render(){
        return(
            <Error>
                <section className='newRequestPage'>
                    <div className='sideBar'>
                        <button className='backButton' onClick={this.props.history.goBack}>Back</button>
                    </div>
                    <div className='mainBar'>
                        <h2>What are you looking for?</h2>
                        <form className='newRequestForm' onSubmit={this.handleSubmit}>
                            <div className='inputBox'>
                                <label htmlFor="product">Product * </label>
                                <input className='newReqInput' type="text" name='product' id='product' required='require'/>
                            </div>
                            <div className='inputBox'>
                                <label htmlFor="category">Category * </label>
                                <select className='newReqInput' id='category' name='category' required>
                                    <option value=''> </option>
                                    <option value='Appliances'>Appliances</option>
                                    <option value='Baby & Kid'>Baby & Kid</option>
                                    <option value='Clothing & Shoes'>Clothing & Shoes</option>
                                    <option value='Entertainment'>Entertainment</option>
                                    <option value='Farm & Garden'>Farm & Garden</option>
                                    <option value='Furniture'>Furniture</option>
                                    <option value='Health & Beauty'>Health & Beauty</option>
                                    <option value='Household'>Household</option>
                                    <option value='Jewelry'>Jewelry</option>
                                    <option value='Materials'>Materials</option>
                                    <option value='Sporting Goods'>Sporting Goods</option>
                                    <option value='Technology'>Technology</option>
                                    <option value='Toys & Games'>Toys & Games</option>
                                    <option value='Transportation'>Transportation</option>
                                    <option value='Other'>Other</option>
                                </select>
                            </div>
                            <div className='inputBox'>
                                <label htmlFor="moreInfo">More Info *</label>
                                <textarea className='newReqInput' rows={10} name='moreInfo' id='moreInfo' required='require' />
                            </div>
                            <input className='submitRequest submitRequestButton' type='submit' value='Submit'/>
                        </form> 
                    </div>
                </section>
            </Error>
        )
    }
}