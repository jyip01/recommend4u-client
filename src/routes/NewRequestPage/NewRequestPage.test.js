import React from 'react'
import ReactDOM from 'react-dom'
import NewRequestPage from './NewRequestPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><NewRequestPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NewRequestPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
