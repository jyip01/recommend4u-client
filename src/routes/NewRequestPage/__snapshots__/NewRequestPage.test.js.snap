// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<section
  className="newRequestPage"
>
  <div
    className="sideBar"
  >
    <button
      className="backButton"
    >
      Back
    </button>
  </div>
  <div
    className="mainBar"
  >
    <h2>
      What are you looking for?
    </h2>
    <form
      className="newRequestForm"
      onSubmit={[Function]}
    >
      <div
        className="inputBox"
      >
        <label
          htmlFor="product"
        >
          Product * 
        </label>
        <input
          className="newReqInput"
          id="product"
          name="product"
          required="require"
          type="text"
        />
      </div>
      <div
        className="inputBox"
      >
        <label
          htmlFor="category"
        >
          Category * 
        </label>
        <select
          className="newReqInput"
          id="category"
          name="category"
          required={true}
        >
          <option
            value=""
          >
             
          </option>
          <option
            value="Appliances"
          >
            Appliances
          </option>
          <option
            value="Baby & Kid"
          >
            Baby & Kid
          </option>
          <option
            value="Clothing & Shoes"
          >
            Clothing & Shoes
          </option>
          <option
            value="Entertainment"
          >
            Entertainment
          </option>
          <option
            value="Farm & Garden"
          >
            Farm & Garden
          </option>
          <option
            value="Furniture"
          >
            Furniture
          </option>
          <option
            value="Health & Beauty"
          >
            Health & Beauty
          </option>
          <option
            value="Household"
          >
            Household
          </option>
          <option
            value="Jewelry"
          >
            Jewelry
          </option>
          <option
            value="Materials"
          >
            Materials
          </option>
          <option
            value="Sporting Goods"
          >
            Sporting Goods
          </option>
          <option
            value="Technology"
          >
            Technology
          </option>
          <option
            value="Toys & Games"
          >
            Toys & Games
          </option>
          <option
            value="Transportation"
          >
            Transportation
          </option>
          <option
            value="Other"
          >
            Other
          </option>
        </select>
      </div>
      <div
        className="inputBox"
      >
        <label
          htmlFor="moreInfo"
        >
          More Info *
        </label>
        <textarea
          className="newReqInput"
          id="moreInfo"
          name="moreInfo"
          required="require"
          rows={10}
        />
      </div>
      <input
        className="submitRequest submitRequestButton"
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</section>
`;
